dhis-2/dhis-services/dhis-service-program-rule/src/main/java/org/hisp/dhis/programrule/api/NotificationEffect.java

```diff
@@ -27,14 +27,13 @@
  */
 package org.hisp.dhis.programrule.api;
 
+import java.util.Date;
 import org.hisp.dhis.common.UID;
 
 /**
  * This effect will be used in tracker to send a notification.
  *
- * @param type the effect type
- * @param rule the UID of the rule that generated this effect
  * @param template the UID of the template of the notification
- * @param date the date when to send the notification. Only needed for `SCHEDULE_MESSAGE` type
+ * @param date the date when to send the notification. It will be sent immediately if missing.
```

Comment:
```suggestion
 * @param date the date when to send the notification. The notification is sent immediately when the date is null.
```

Missing sounds like a mistake IMHO